README.md

```diff
@@ -6,26 +8,114 @@ An awesome tool to develop and test your micro-frontend services !
 
 Install
 -------
-`$ npm install -g @capsulajs/capsula-hub`
-
-or
-
-`$ yarn global add @capsulajs/capsula-hub`
+In your project, run `$ npm install @capsulajs/capsula-hub` or `$ yarn add @capsulajs/capsula-hub`.
+If you want to use `capsula-hub` command directly, you can install it globally.
 
 Usage
 -----
-`capsula-hub run` if you install it globally
-`yarn/npm run` with npm scripts
+`capsula-hub run -l <path/to/configuration_file>`
+
+This command will run the app with the provided configuration.
+
+Configuration
+-------------
+The configuration should be a `js` file that exports an `object` which
+match this [API](https://github.com/capsulajs/capsulahub-core/blob/develop/packages/workspace/src/api/WorkspaceConfig.ts).
+
+Example: _config.js_
+```javascript
+module.exports = {
+  name: 'my-app',
+  service: [
+    {
+      serviceName: 'myServiceA',
+      path: 'https://my-cdn/my-service-a',
+      definition: {
+        serviceName: 'myServiceA',
+        methods: {
+          myMethod1: { asyncModel: 'RequestResponse' },
+          myMethod2$: { asyncModel: 'RequestStream' },
+        }
+      },
+      config: {}
+    },
+    {
+      serviceName: 'myServiceB',
+      path: 'https://my-cdn/my-service-b',
+      definition: {
+        serviceName: 'myServiceB',
+        methods: {
+          myMethod1: { asyncModel: 'RequestResponse' },
+          myMethod2: { asyncModel: 'RequestResponse' },
+        }
+      },
+      config: {}
+    },
+  ],
+  components: {
+    layouts: {
+      grid: {
+        componentName: 'my-grid',
+        path: 'http://my-cdn/components/Grid.js',
+        config: { title: 'Base Grid' },
+       },
+    },
+    items: {
+      myItem: {
+        componentName: 'my-item',
+        path: 'http://my-cdn/components/myItem.js',
+        config: { title: 'Base Item' },
+      },
+    }
+  }
+};
+```
+
+Develop your extension
+----------------------
 
+An extension is a service or a web component that is loaded by CapsulaHub. 
+The extension should look like this:
+```typescript
+import { Workspace } from '@capsulajs/capsulahub-core';
+export default (workspace: Workspace, config: any): Promise<void> => {
+  // your code here
+};
 
+```
+`workspace` and `config` are injected by the application in the extension.
+
+`workspace` is matching this [API](https://github.com/capsulajs/capsulahub-core/blob/develop/packages/workspace/src/api/Workspace.ts) 
+and could be use in particular to allow the service to register itself.
+
+`config` contains the configuration for this specific extension that you passed in the configuration file 
+(e.g: `configuration_file.services.['myExtension'].config` for myExtension service).
+
+For more details, take a look at this 
+[example file](https://github.com/capsulajs/capsulahub-core/blob/develop/packages/externalModules/src/services/serviceA.ts).
+
```

Comment:
I think we should mention that a user has to call workspace.registerService inside his service bootstrap (and DON'T wait for this request to be resolved before resolving his bootstrap promise).

Reply:
We will detail this more in [this repo](https://github.com/capsulajs/capsulahub-extensions)